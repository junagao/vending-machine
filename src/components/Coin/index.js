export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
