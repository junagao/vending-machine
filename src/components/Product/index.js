export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
