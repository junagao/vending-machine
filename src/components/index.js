export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
