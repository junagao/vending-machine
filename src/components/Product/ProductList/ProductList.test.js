import React from 'react';
import { shallow } from 'enzyme';
import { ProductList, ProductItem } from 'components';

describe('ProductList', () => {
  let wrapper;

  const productsMock = [
    {
      id: 'coke',
      name: 'coke',
      img: 'image/coke.png',
      price: 1,
      currentStock: 10,
    },
    {
      id: 'sprite',
      name: 'sprite',
      img: 'image/sprite.png',
      price: 1.25,
      currentStock: 5,
    },
    {
      id: 'fanta',
      name: 'fanta',
      img: 'image/fanta.png',
      price: 1.25,
      currentStock: 11,
    },
  ];

  const onSelectProductMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <ProductList
        products={productsMock}
        onSelectProduct={onSelectProductMock}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one unordered list ProductListUl element', () => {
    expect(wrapper.find('ProductListUl').length).toEqual(1);
  });

  it('should render 3 instances of <ProductItem />', () => {
    expect(wrapper.find(ProductItem).length).toEqual(3);
  });
});
