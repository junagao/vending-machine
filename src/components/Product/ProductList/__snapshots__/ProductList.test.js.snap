// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should render correctly given the required props 1`] = `
<ProductListUl>
  <ProductItem
    currentStock={10}
    id="coke"
    img="image/coke.png"
    key="coke"
    name="coke"
    onSelectProduct={[MockFunction]}
    price={1}
  />
  <ProductItem
    currentStock={5}
    id="sprite"
    img="image/sprite.png"
    key="sprite"
    name="sprite"
    onSelectProduct={[MockFunction]}
    price={1.25}
  />
  <ProductItem
    currentStock={11}
    id="fanta"
    img="image/fanta.png"
    key="fanta"
    name="fanta"
    onSelectProduct={[MockFunction]}
    price={1.25}
  />
</ProductListUl>
`;
